src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncLocalsTests.cs

```diff
@@ -1273,5 +1273,263 @@ public class C
                 }, module.GetFieldNames("C.<F>d__0"));
             });
         }
+
+        [WorkItem(9775, "https://github.com/dotnet/roslyn/issues/9775")]
+        [Fact]
+        public void Fixed_Debug()
+        {
+            var text =
+@"using System;
+using System.Threading.Tasks;
+class C
+{
+    static async Task<int> F(byte[] b)
+    {
+        int i;
+        unsafe
+        {
+            fixed (byte* p = b)
+            {
+                i = *p;
+            }
+        }
+        await Task.Yield();
+        return i;
+    }
+    static void Main()
+    {
+        var i = F(new byte[] { 1, 2, 3 }).Result;
+        Console.Write(i);
+    }
+}";
+            var verifier = CompileAndVerify(text, options: TestOptions.UnsafeReleaseExe, expectedOutput: @"1");
```

Comment:
>TestOptions.UnsafeReleaseExe [](start = 59, length = 28)

It looks like this test option uses OptimizationLevel.Release and the modified code is not used for this option. Consider also testing with OptimizationLevel that would cause compiler to take the modified code path.